factory/wsSovereignNotifierFactory.go

```diff
@@ -14,5 +18,35 @@ func CreatWsSovereignNotifier(cfg config.Config) (process.WSClient, error) {
 		return nil, err
 	}
 
-	return wsclient.NewWsClient(sovereignNotifier)
+	dataIndexer, err := indexer.NewIndexer(sovereignNotifier)
+	if err != nil {
+		return nil, err
+	}
+
+	marshaller, err := factory.NewMarshalizer(cfg.WebSocketConfig.MarshallerType)
+	if err != nil {
+		return nil, err
+	}
+
+	operationHandler, err := indexer.NewOperationHandler(dataIndexer, marshaller)
+	if err != nil {
+		return nil, err
+	}
+
+	uint64ByteSliceConverter := uint64ByteSlice.NewBigEndianConverter()
+	payloadParser, err := websocketOutportDriver.NewWebSocketPayloadParser(uint64ByteSliceConverter)
+	if err != nil {
+		return nil, err
+	}
+
+	argsWsClient := &client.ArgsWsClient{
+		Url:                      cfg.WebSocketConfig.Url,
+		RetryDurationInSec:       cfg.WebSocketConfig.RetryDuration,
+		BlockingAckOnError:       false,
```

Comment:
I would also take this from the configs.

Reply:
Right, good suggestion, applied it